custom_components/wyzeapi/sensor.py

```diff
@@ -215,3 +244,219 @@ def extra_state_attributes(self):
     @property
     def native_value(self):
         return self._camera.device_params.get("electricity")
+
+
+class WyzePlugEnergySensor(RestoreSensor):
+    """Respresents an Outdoor Plug Total Energy Sensor."""
+
+    _attr_device_class = SensorDeviceClass.ENERGY
+    _attr_native_unit_of_measurement = UnitOfEnergy.KILO_WATT_HOUR
+    _attr_state_class = SensorStateClass.TOTAL_INCREASING
+    _attr_suggested_display_precision = 3
+    _previous_hour = None
+    _previous_value = None
+    _past_hours_previous_value = None
+    _current_value = 0
+    _past_hours_value = 0
+    _hourly_energy_usage_added = 0
+
+    def __init__(
+        self, switch: Switch, switch_usage_service: SwitchUsageService
+    ) -> None:
+        """Initialize an energy sensor."""
+        self._switch = switch
+        self._switch_usage_service = switch_usage_service
+        self._switch.usage_history = None
+
+    @property
+    def name(self) -> str:
+        """Get the name of the sensor."""
+        return "Total Energy Usage"
+
+    @property
+    def unique_id(self):
+        """Get the unique ID of the sensor."""
+        return f"{self._switch.nickname}.energy-{self._switch.mac}"
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self._switch.mac)},
+            "name": self._switch.nickname,
+        }
+
+    def update_energy(self):
+        """Update the energy sensor."""
+        _now = int(datetime.utcnow().hour)
+        self._hourly_energy_usage_added = 0
+
+        if self._switch.usage_history:
+            _raw_data = self._switch.usage_history
+            _LOGGER.debug(_raw_data)
+            _current_day_list = json.loads(_raw_data[0]["data"])
+
+            if len(_raw_data) > 0:  # Confirm there is data
+                if _now == 0:  # Handle rolling to the next UTC day
+                    self._past_hours_value = _current_day_list[23] / 1000
+                    if len(_raw_data) > 1:  # New Day's value
```

Comment:
This is the only other scenario, around mightnight UTC there are 2 lists in the data block, 1 for yesterday and another for the following day which will be mostly 0's except for possibly the 00Z hour. Let me know if you have questions. I'll say I've had this running for probably close to a year now 😆 with no issues. But happy to make changes where you see fit.

Reply:
Oh gotcha!!